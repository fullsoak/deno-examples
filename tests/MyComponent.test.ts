import { ssr } from "fullsoak";
import { MyComponent } from "../src/components/MyComponent/index.tsx";
import { assertSnapshot } from "@std/testing/snapshot";

Deno.test("MyComponent", async (t) => {
  const output = await ssr(MyComponent, { foo: "bar" });
  await assertSnapshot(t, output);
});
