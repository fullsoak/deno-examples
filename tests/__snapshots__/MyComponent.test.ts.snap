export const snapshot = {};

snapshot[`MyComponent 1`] = `
'<!doctype html><html><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0"><title></title><script type="importmap">
{
  "imports": {
    "preact": "https://esm.sh/preact@10.25.4",
    "preact/hooks": "https://esm.sh/preact@10.25.4/hooks",
    "react": "https://esm.sh/preact@10.25.4",
    "react/jsx-runtime": "https://esm.sh/preact@10.25.4/jsx-runtime",
    "preact/jsx-runtime": "https://esm.sh/preact@10.25.4/jsx-runtime",
    "react-dom": "https://esm.sh/preact@10.25.4/compat/",
    "react-dom/*": "https://esm.sh/preact@10.25.4/compat/*",
    "htm/preact": "https://esm.sh/htm@3.1.1/preact?external=preact",
    "preact-iso": "https://esm.sh/preact-iso?external=preact",
    "fullsoak/preact-iso": "https://esm.sh/preact-iso?external=preact",
    "@fullsoak/fullsoak/preact-iso": "https://esm.sh/preact-iso?external=preact",
    "fullsoak": "/fullsoak",
    "@fullsoak/fullsoak": "/fullsoak"
  }
}
</script><script type="text/javascript"></script><style></style></head><body><main id="fullsoak-htmlshell-main"><section class="main"><h1>MyComponent</h1><ul><li>props &#39;foo&#39; is bar</li><li>state count is 0 <button >+</button><button >-</button></li><li><a href="/app">Route Aware App Component</a></li></ul></section><section style="padding:1rem;"><h2>MyOtherComponent</h2><span>props &#39;baz&#39; is 0</span></section><footer><span>source code of this website:</span> <a href="https://github.com/fullsoak/deno-examples">https://github.com/fullsoak/deno-examples</a></footer></main><script type="text/javascript">window.preloadedProps = {"foo":"bar"}</script><script type="module" src="/js/MyComponent/mount.js"></script></body></html>'
`;
